engine/graphics/src/webgpu/graphics_webgpu_private.h

```diff
@@ -24,218 +24,223 @@
+        uint32_t m_Height;
+        uint32_t m_OriginalWidth;
+        uint32_t m_OriginalHeight;
+        uint16_t m_Depth;
+        uint16_t m_TextureSamplerIndex : 10;
+        uint16_t m_MipMapCount : 5;
+        uint8_t m_UsageHintFlags;
+        uint8_t m_Destroyed : 1;
+        uint8_t : 8; // aligned by 32 bit
     };
 
     struct WebGPURenderTarget
     {
-        WebGPURenderTarget() {
+        WebGPURenderTarget()
+        {
             memset(this, 0, sizeof(*this));
         }
 
-        AttachmentOp   m_ColorBufferLoadOps[MAX_BUFFER_COLOR_ATTACHMENTS];
-        AttachmentOp   m_ColorBufferStoreOps[MAX_BUFFER_COLOR_ATTACHMENTS];
-        float          m_ColorBufferClearValue[MAX_BUFFER_COLOR_ATTACHMENTS][4];
+        AttachmentOp m_ColorBufferLoadOps[MAX_BUFFER_COLOR_ATTACHMENTS];
+        AttachmentOp m_ColorBufferStoreOps[MAX_BUFFER_COLOR_ATTACHMENTS];
+        float m_ColorBufferClearValue[MAX_BUFFER_COLOR_ATTACHMENTS][4];
 
-        HTexture       m_TextureResolve[MAX_BUFFER_COLOR_ATTACHMENTS];
-        HTexture       m_TextureColor[MAX_BUFFER_COLOR_ATTACHMENTS];
-        HTexture       m_TextureDepthStencil;
+        HTexture m_TextureResolve[MAX_BUFFER_COLOR_ATTACHMENTS];
+        HTexture m_TextureColor[MAX_BUFFER_COLOR_ATTACHMENTS];
+        HTexture m_TextureDepthStencil;
 
-        uint32_t       m_Width;
-        uint32_t       m_Height;
-        float          m_Scissor[4];
-        uint8_t        m_Multisample;
-        uint32_t       m_ColorBufferCount : 7;
+        float m_Scissor[4];
+        uint32_t m_Width;
+        uint32_t m_Height;
+        uint8_t m_Multisample;
+        uint32_t m_ColorBufferCount : 7;
+        uint32_t : 17;
     };
 
     struct WebGPUComputePass
     {
         WGPUComputePassEncoder m_Encoder;
-        WGPUComputePipeline    m_Pipeline;
+        WGPUComputePipeline m_Pipeline;
     };
 
     struct WebGPURenderPass
     {
-        WebGPURenderTarget*    m_Target;
-        WGPURenderPassEncoder  m_Encoder;
-        WGPURenderPipeline     m_Pipeline;
-        WGPUBindGroup          m_BindGroups[MAX_SET_COUNT];
-        WebGPUBuffer*          m_VertexBuffers[MAX_VERTEX_BUFFERS];
-        WebGPUBuffer*          m_IndexBuffer;
+        WGPUBindGroup m_BindGroups[MAX_SET_COUNT];
+        WebGPUBuffer* m_VertexBuffers[MAX_VERTEX_BUFFERS];
+        WebGPURenderTarget* m_Target;
+        WGPURenderPassEncoder m_Encoder;
+        WGPURenderPipeline m_Pipeline;
+        WebGPUBuffer* m_IndexBuffer;
     };
 
     struct WebGPUTextureSampler
     {
-        WGPUSampler   m_Sampler;
+        WGPUSampler m_Sampler;
         TextureFilter m_MinFilter;
         TextureFilter m_MagFilter;
-        TextureWrap   m_AddressModeU;
-        TextureWrap   m_AddressModeV;
-        float         m_MaxAnisotropy;
-        uint8_t       m_MaxLod;
+        TextureWrap m_AddressModeU;
+        TextureWrap m_AddressModeV;
+        float m_MaxAnisotropy;
+        uint8_t m_MaxLod;
+        uint32_t : 24;
```

Comment:
Try align structure.

Reply:
What's the background? Compiler warnings?

Reply:
Struct size to be multiple of 32.

Reply:
Removed.